apis/v1alpha2/validation/gateway.go

```diff
@@ -26,6 +27,11 @@ import (
 	gatewayv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
 )
 
+var (
+	tcpUdpProtocols = []gatewayv1a2.ProtocolType{gatewayv1a2.TCPProtocolType, gatewayv1a2.UDPProtocolType}
+	nonTlsProtocols = []gatewayv1a2.ProtocolType{gatewayv1a2.HTTPProtocolType, gatewayv1a2.UDPProtocolType, gatewayv1a2.TCPProtocolType}
+)
```

Comment:
Since these are only used for `isProtocolInSubset`, I'd make them a set, eg:
```Go
var map[ProtocolType]struct{} = map[ProtocolType]struct{}{
        gatewayv1a2.HTTPProtocolType: struct{}{},
        gatewayv1a2.UDPProtocolType: struct{}{},
        gatewayv1a2.TCPProtocolType: struct{}{},
}

```

You could take it a bit further to use a convenience type, but not sure if that's worth the effort:

```Go
type ProtocolSubset map[gatewayv1a2.ProtocolType]struct{}

func (p ProtocolSubset) Contains(proto gatewayv1a2.ProtocolType) bool {
        _, ok := p[proto]
        return ok
}
```

Reply:
I think making a map is a nice idea here . Making a separate convenicence type I am not sure about that . Uploading the patch for the map

Reply:
As a note, this is very familiar to the widely used k8s set utils: https://github.com/kubernetes/apimachinery/blob/master/pkg/util/sets/string.go.